### PR TITLE
[Modules] Update SQL MI to new dependencies approach

### PR DESCRIPTION
**Pull request by eriqua**
# Description

- Converting to new dependencies approach
- Fixed dependency between keys and encryption object

| Pipeline |
| - |
| [Sql: ManagedInstances](https://github.com/Azure/ResourceModules/actions/workflows/ms.sql.managedinstances.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: modules/Microsoft.Sql/managedInstances/.test/common/dependencies.bicep [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+@description('Required. The name of the Managed Identity to create.')
+param managedIdentityName string
+
+@description('Required. The name of the Key Vault to create.')
+param keyVaultName string
+
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+var sqlMiVnetAddressPrefix = '10.0.0.0/16'
```
Comment (AlexanderSehr) [inline modules/Microsoft.Sql/managedInstances/.test/common/dependencies.bicep:19]: why the variable(s) if they're only used in one single location?

File: modules/Microsoft.Sql/managedInstances/.test/min/dependencies.bicep [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+@description('Required. The name of the Network Security Group to create.')
+param networkSecurityGroupName string
+
+@description('Required. The name of the Route Table to create.')
+param routeTableName string
+
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+var sqlMiVnetAddressPrefix = '10.0.0.0/16'
```
Comment (AlexanderSehr) [inline modules/Microsoft.Sql/managedInstances/.test/min/dependencies.bicep:13]: Same as the other comment. If those variables are used only in a single location, I'd much rather put them there like all the other values

Comment (eriqua) [reply modules/Microsoft.Sql/managedInstances/.test/min/dependencies.bicep:13]: The subnet prefix is widely used. The vnet prefix is related to the subnet one. If for any reason one would need to change their value, they would likely both be updated, so I suggest to keep them close.

File: modules/Microsoft.Sql/managedInstances/.test/common/dependencies.bicep (hunk 1 above)
Comment (eriqua) [reply modules/Microsoft.Sql/managedInstances/.test/common/dependencies.bicep:19]: Same as other comment

File: modules/Microsoft.Sql/managedInstances/.test/min/dependencies.bicep (hunk 2 above)
Comment (AlexanderSehr) [reply modules/Microsoft.Sql/managedInstances/.test/min/dependencies.bicep:13]: Hm OK. Personally, I feel it is confusing to introduce values in other places if not absolutely necessary, but I won't block the PR for it.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**